src/cli/java/org/commcare/util/screen/SyncScreen.java

```diff
@@ -56,64 +49,30 @@ private void parseMakeRequest() throws CommCareSessionException {
         }
 
         try {
-            Response response = makeSyncRequest(syncPost);
-            if (!response.isSuccessful()) {
-                printStream.println(String.format("Sync request failed with response code %s and message %s", response.code(), response.body()));
-                printStream.println("Press 'enter' to retry.");
-                return;
-            }
+            int responseCode = SessionUtils.doPostRequest(
+                syncPost, sessionWrapper, username, password, printStream
+            );
             syncSuccessful = true;
-            if (response.code() != 204) {
+            if (responseCode != 204) {
                 SessionUtils.restoreUserToSandbox(sessionWrapper.getSandbox(),
                     sessionWrapper,
                     sessionWrapper.getPlatform(),
                     username,
                     password,
                     printStream);
 
-                printStream.println(String.format("Sync successful with response %s", response));
+                printStream.println("Sync successful with response");
             } else {
                 printStream.println("Did not sync because case was already claimed.");
             }
             printStream.println("Press 'enter' to continue.");
-        } catch (IOException e) {
+        } catch (Exception e) {
```

Comment:
I think there is value in keeping this as `IOException` so that we don't gulp any code errors like NPE. `IOException` is more anticipated because of a network request involved and we should throw the underlying exception on unsuccesful response code as `IOException` as well.

Reply:
8fc8d2d02